### ISSUE TITLE
A tiny bug in http://keras.io/backend/#batch_dot

### ISSUE THREAD

**Issue by suquark** [fchollet/keras#3141]
In `example` section of http://keras.io/backend/#batch_dot, it reads:

Assume x = [[1, 2] and y = [[5, 6] [3, 4]] [7, 8]] batch_dot(x, y, axes=1) = [[17, 53]] ...

It confused me for a while and later I refer to the comment of source (https://github.com/fchollet/keras/blob/master/keras/backend/theano_backend.py, L130), 
and it should look like:
```
Assume x = [[1, 2]   and y = [[5, 6]
            [3, 4]]          [7, 8]]
```
I think mkdocs just views it as two lines and then combine them into a single line, so matrix should be written in a single line. 
May you fix it and notice other potential bugs like this. Thanks.

**Comment by EderSantana** [comment]
@suquark thanks for letting us know, please check the PR with the changes your suggested

**Comment by alyato** [comment]
@suquark ，I don't understand what the batch_dot do?

> batch_dot results in a tensor with less dimensions than the input. If the number of dimensions is reduced to 1, we use expand_dims to make sure that ndim is at least 2

what  does the axes mean? Can it be set 0 ,1,2?

>          x = [[1, 2]   and y = [[5, 6]
            [3, 4]]          [7, 8]]

The value is  [[17, 53]]. how to get the value?
I check the dot function.

> Multiplies 2 tensors. When attempting to multiply a ND tensor with a ND tensor, reproduces the Theano behavior (e.g. (2, 3).(4, 3, 5) = (2, 4, 5))

it is also confused me ? Do you give me  some suggestions.

Thanks.

**Comment by alyato** [comment]
@suquark i check the #3142,and find one example.

>            
          Assume x = [[1, 2], [3, 4]]   and y = [[5, 6], [7, 8]]
          batch_dot(x, y, axes=1) = [[17, 53]] which is the main diagonal
          of x.dot(y.T), although we never have to calculate the off-diagonal
          elements.

if axes = 1,the result is the main diagonal of x.dot(y.T).
Assume axes = 0,or 2,how to understand?
Thanks.

**Comment by suquark** [comment]
@alyato 

For dot,
> Multiplies 2 tensors. When attempting to multiply a ND tensor with a ND tensor, reproduces the Theano behavior (e.g. (2, 3).(4, 3, 5) = (2, 4, 5))

, it really means

> Assuming tensor A with shape (a,**k**) and B with (b,**k**,c), A.dot(B) gets a tensor with shape (a,b,c), where **k** can be considered as 'contracted'.

It should be viewed as an expansion of matrix product (dot between matrices). Matrix A has shape (a,**b**), and B has shape (**b**,c), A dot B gets C with shape (a,c). Now a tensor may have more dims than a matrix, but it just did in the same way:
```
C[a,b,c,...,r,t,u,v,w,...,z] = \sum_s A[a,b,c,...,s,t]*B[u,v,w,...,z,s]
```
Of course, vectors(ndim=1) are dealt in a special way for convenience.


For batch dot,
axes means the index of the subscript in a tensor (since a tensor can be viewed as a multi-dim array). 
batch_dot means make batch from an axes and dot.  In this case, 
x is batched into (1)  \[1, 2]  (2) [3, 4]
y  is batched into (1) \[5, 6]  (2) [7, 8]
and get a result from each batch, so [1,2]*[5,6]=17, [3,4]*[7,8]=53
finally concat the result from each batch and get [17, 53]

if you make batch from axes 0,  x=>[1,3],[2,4] , ... and you will get different results.

I hope it is useful to you.

**Comment by alyato** [comment]
@suquark yeah, i got it. Thanks
